spec/pry_spec.rb

```diff
@@ -184,6 +184,21 @@ class Hello
           .to raise_error SignalException
       end
 
+      it "should return if started inside signal handler" do
+        if Pry::Helpers::Platform.jruby?
+          skip "jruby allows mutex usage in signal handlers"
+        end
+
+        if Gem::Version.new(RUBY_VERSION) < Gem::Version.new("2.0.0")
+          skip "pre-2.0 mri allows mutex usage in signal handlers"
+        end
+
+        output = nil
+        trap("USR1") { output = mock_pry }
```

Comment:
As a cleanup step, I would nullify this hook like so:

```suggestion
        trap("USR1") do
           # do nothing
        end
```

This way if another test decides to `kill` with `USR1`, we wouldn't invoke this code. IIRC, there's no other way to remove the hook.

P.S. Rubocop might complain about empty block, I would just silence the cop in that case.

Reply:
Good idea, updated.